azure-pipelines-regression-test.yml

```diff
@@ -0,0 +1,309 @@
+    - .github/*
+    - '*.md'
+
+pr:
+  branches:
+    include:
+    - main
+    - release/*
+    - feature/*
+  paths:
+    include:
+    - src/Compiler/*
+    - src/fsc/*
+    - src/FSharp.Core/*
+    - src/FSharp.Build/*
+    exclude:
+    - docs/*
+    - .github/*
+    - '*.md'
+
+variables:
+  - name: _TeamName
+    value: FSharp
+  - name: _BuildConfig
+    value: Release
+  - name: _PublishUsingPipelines
+    value: true
+  - name: Codeql.Enabled
+    value: false  # Disabled for regression tests
+  # Pick up pool provider name behavior from shared yaml template
+  - template: /eng/common/templates/variables/pool-providers.yml
+
+# Matrix of third-party repositories to test against
+# This configuration can be extended in the future
+parameters:
+- name: testMatrix
+  type: object
+  default:
+  - repo: fsprojects/FSharpPlus
+    commit: f614035b75922aba41ed6a36c2fc986a2171d2b8
+    buildScript: build.cmd
+    displayName: FSharpPlus
+
+stages:
+- stage: RegressionTest
+  displayName: F# Compiler Regression Tests
+  jobs:
+  - job: BuildCompiler
+    displayName: Build F# Compiler
+    pool:
+      name: $(DncEngPublicBuildPool)
+      demands: ImageOverride -equals windows.vs2022.amd64.open
+    timeoutInMinutes: 60
+    steps:
+    - checkout: self
+      clean: true
+      displayName: Checkout F# compiler source
+
+    - script: dotnet --list-sdks
+      displayName: Report .NET SDK versions
+
+    # We first download a publicly available .NET SDK. That one has support for `path` in global.json. 
+    # dotnet.cmd script can then download a version which is not yet shipped, but matches global.json.
+    - task: UseDotNet@2
+      displayName: install SDK
+      inputs:
+        packageType: sdk
+        version: '10.x'
+        includePreviewVersions: true
+        workingDirectory: $(Build.SourcesDirectory)
+        installationPath: $(Build.SourcesDirectory)/.dotnet
+
+    - script: .\eng\common\dotnet.cmd
+      displayName: Ensure correct .NET SDK version
+
+    - script: .\Build.cmd -c $(_BuildConfig) -pack
+      env:
+        NativeToolsOnMachine: true
+      displayName: Build F# compiler and create packages
+
+    - task: PublishPipelineArtifact@1
+      displayName: Publish F# Compiler Artifacts
+      inputs:
+        targetPath: '$(Build.SourcesDirectory)/artifacts'
+        artifactName: 'FSharpCompilerArtifacts'
+        publishLocation: pipeline
+      condition: succeeded()
+
+    - task: PublishPipelineArtifact@1
+      displayName: Publish UseLocalCompiler props file
+      inputs:
+        targetPath: '$(Build.SourcesDirectory)/UseLocalCompiler.Directory.Build.props'
+        artifactName: 'UseLocalCompilerProps'
+        publishLocation: pipeline
+      condition: succeeded()
+
+  # Test against third-party repositories
+  - ${{ each item in parameters.testMatrix }}:
+    - job: Test_${{ replace(item.repo, '/', '_') }}
+      displayName: 'Regression Test: ${{ item.displayName }}'
+      dependsOn: BuildCompiler
```

Comment:
Should depend on existign `EndToEndBuildTests` which also builds the compiler, just needs to emit the artifacts. Read here: https://github.com/dotnet/fsharp/blob/main/azure-pipelines-PR.yml